torch/csrc/jit/passes/lower_grad_of.cpp

```diff
@@ -19,7 +19,7 @@ void LowerGradOf(Graph& g) {
       if_stat->addBlock()->cloneFrom(
           it->blocks().at(0), [](Value* v) { return v; });
       auto else_block = if_stat->addBlock();
-      auto undef = g.createUndefined()
+      auto undef = g.createNone(TensorType::get())
```

Comment:
Also seems unsafe: computations that use the output of this if_block will expect the type to be tensor, but now an Optional[Tensor] type has been introduced.